### DISCUSSION TITLE
How do I take the partial derivative (Using FourDivergence) of a fraction with respect to a perpendicular light cone component?

### DISCUSSION THREAD

**Discussion by Dheepan01** [FeynCalc/feyncalc#226] [General]
Lets say I have a function of p1, f[p1] which I need to differentiate with respect to the perpendicular component of p1 with the lorentz index nu. The image attached shows this situation.

The problem with this is because the normal derivative of p1 would give the metric tensor, but here it should be the metric tensor with a perp component, and the perp symbol causes the error 'Is not a lorentz vector' for Four Divergence.

Can someone advice me on how to do this? (the other image attached is what you would get if you differentiate by hand)

<img width="170" alt="answer" src="https://github.com/FeynCalc/feyncalc/assets/100049600/283344ea-88f3-42c0-8af2-e3ee72331110">

<img width="164" alt="question" src="https://github.com/FeynCalc/feyncalc/assets/100049600/3a476af7-4605-499d-a7a9-2e5447c74b66">

**Comment by vsht** [comment]
See here

https://feyncalc.github.io/FeynCalcBookDev/LightCone.html
